client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -0,0 +1,144 @@
+			isDismissModalOpen: false,
+		};
+	}
+
+	componentDidMount() {
+		this.trackBannerEvent( 'shipping_banner_show' );
+	}
+
+	closeDismissModal = () => {
+		this.setState( { isDismissModalOpen: false } );
+		this.trackBannerEvent(
+			'shipping_banner_dismiss_modal_close_button_click'
+		);
+	};
+
+	openDismissModal = () => {
+		this.setState( { isDismissModalOpen: true } );
+		this.trackBannerEvent( 'shipping_banner_dimiss_click' );
+	};
+
+	hideBanner = () => {
+		this.setState( { showShippingBanner: false } );
+	};
+
+	createShippingLabelClicked = () => {
+		// TODO: install and activate WCS
+		// TODO: open WCS modal
+		this.trackBannerEvent( 'shipping_banner_create_label_click' );
+	};
+
+	learnMoreClicked = () => {
+		this.trackBannerEvent( 'shipping_banner_learn_more_click' );
+	};
+
+	trackBannerEvent = ( eventName ) => {
+		recordEvent( eventName, {
+			jetpack_installed: this.activePlugins.includes( 'jetpack' ),
+			jetpack_connected: this.isJetpackConnected,
+			wcs_installed: this.activePlugins.includes(
+				'woocommerce-services'
+			),
+		} );
+	};
+
+	render() {
+		const { isDismissModalOpen, showShippingBanner } = this.state;
+		const { itemsCount } = this.props;
+		if ( ! showShippingBanner ) {
+			return null;
+		}
+
+		return (
+			<div>
+				<div className="wc-admin-shipping-banner-container">
+					<img
+						className="wc-admin-shipping-banner-illustration"
+						src={ wcAdminAssetUrl + 'shippingillustration.svg' }
+						alt={ __( 'Shipping ', 'woocommerce-admin' ) }
+					/>
+					<h3>
+						{ sprintf(
+							_n(
+								'Fulfill %d item with WooCommerce Shipping',
+								'Fulfill %d items with WooCommerce Shipping',
+								itemsCount,
+								'woocommerce-admin'
+							),
+							itemsCount
+						) }
+					</h3>
+					<p>
+						{ __(
+							'Print discounted shipping labels with a click. This will install WooCommerce Services. '
+						) }
+						<ExternalLink
+							href="woocommerce.com"
+							onClick={ this.learnMoreClicked }
+						>
+							{ __( 'Learn More', 'woocommerce-admin' ) }
+						</ExternalLink>
+					</p>
+					<Button
+						isPrimary
+						onClick={ this.createShippingLabelClicked }
+					>
+						{ __( 'Create shipping label' ) }
+					</Button>
+					<button
+						onClick={ this.openDismissModal }
+						type="button"
+						className="notice-dismiss"
+					>
+						<span className="screen-reader-text">
+							{ __(
+								'Close Print Label Banner.',
+								'woocommerce-admin'
+							) }
+						</span>
+					</button>
+				</div>
+				<DismissModal
+					visible={ isDismissModalOpen }
+					onClose={ this.closeDismissModal }
+					onCloseAll={ this.hideBanner }
+					trackBannerEvent={ this.trackBannerEvent }
+				/>
+			</div>
+		);
+	}
+}
+
+export default compose(
+	withSelect( ( select ) => {
+		const { getActivePlugins, isJetpackConnected } = select( 'wc-api' );
+		return {
+			activePlugins: getActivePlugins(),
```

Comment:
This should work okay too without preloading, but if you're worried about a user clicking before this is loaded, we could preload the data outside of onboarding so it's available here as well.